packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/MrwSerializationTypeProvider.cs

```diff
@@ -500,7 +533,20 @@ private MethodBodyStatement[] BuildDeserializationMethodBody()
             MethodBodyStatement rawDataDictionaryDeclaration = MethodBodyStatement.Empty;
             MethodBodyStatement assignRawData = MethodBodyStatement.Empty;
 
-            if (_rawDataField != null)
+            // recursively get the raw data field from myself and all the base
+            var rawDataField = _rawDataField;
+            var baseSerialization = _baseSerializationProvider;
+            while (rawDataField == null)
```

Comment:
can you explain why a `while` is needed here ? I'm having trouble understanding why we chose to use it unless I'm missing something.

Reply:
First please note that a model may not have a raw data if it has a base model because the raw data field will be `private protected` so that derived types could inherit this.

This part is trying to first find if myself defines a raw data field, and if no, find my base (serialization) and see if there is a raw data field, and if none, taking another base - until we have a raw data or there is no more base type.

Reply:
You might also notice that I updated the logic in generating the `DeserializeXXX` method - this is because in our current implementation, this is the only method that we could not delegate.
In this method, the model must handle the JsonDocument for its entire content, including those properties inherited from base (and its base if any), therefore for properties, we could not just use `_model.Properties`, instead we need to append all the properties from its direct base and indirect base.

Same thing happens to raw data, in the entire inheritance tree, only one or none raw data exists, we need find that, and use it here.